File: src/main/scala/com/zh/scala/s3_class/Person1.scala
package com.zh.scala.s3_class

/**
  * 定义类
  * 2019/3/7 22:29
  */
class c1_Person1 {

  var name: String = _ //姓名, _ 表示未赋值

  var age: Int = _ //年龄

}

/**
  * 定义单例类
  */
object person_Test extends App {
  val p = new c1_Person1
  p.name = "zhanghe"
  p.age = 18

  println(p.name + " " + p.age)

}

File: src/main/scala/com/zh/scala/s10_comparable/Person1.java
package com.zh.scala.s10_comparable;

/**
 * @Author zhanghe
 * @Desc:
 * @Date 2019/3/14 21:01
 */
public class Person1 implements Comparable<Person1> {

    private String name;
    private int age;

    @Override
    public int compareTo(Person1 o) {
        return this.age - o.age;  //升序
    }

    public Person1(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }
}
